pgmpy/datasets/_registry.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Set, Type, Union
+
+
+# Exceptions ---------------------------------------------------------
+class DatasetRegistrationError(Exception):
+    """Raised for problems during dataset registration (duplicate names, invalid attrs)."""
+
+
+class DatasetNotFoundError(KeyError):
+    """Raised when requested dataset name is not found in registry."""
+
```

Comment:
I would avoid defining new custom errors unless absolutely necessary. I think we should be able to handle both these scenarios with just `ValueError` and a good error message.